src/charm.py

```diff
@@ -145,24 +175,44 @@ def on_update_status(self, event):
     #        PEER RELATION HOOK HANDLERS         #
     ##############################################
 
-    # hooks: cluster-relation-changed, cluster-relation-departed
+    # Handles relation-changed and relation-departed events
     def reconfigure(self, event):
+        """Reconfigure replicat set
+
+        The number of replicas in the MongoDB replica set is updated.
+        """
         logger.debug("Running reconfigure")
 
         if (
             self.unit.is_leader()
-            and self.state.mongodb_initialized
             and self.need_replica_set_reconfiguration()
         ):
-            self.mongo.reconfigure_replica_set(self.cluster_hosts)
+            try:
+                self.mongo.reconfigure_replica_set(self.cluster_hosts)
+            except Exception as e:
```

Comment:
Can this be a more explicit exception? I think it's fine either way though.

Reply:
You mean specify which exceptions to catch. If so I can try. But the intention was to trap any exception and for any exception type, to ensure that it is raised again so that the event can be deferred.

Reply:
If there are multiple exceptions that could occur, then I don't think it's worth worrying about.